src/sp/it/pl/util/async/executor/ExecuteN.kt

```diff
@@ -0,0 +1,23 @@
+package sp.it.pl.util.async.executor
+
+import java.util.concurrent.Executor
+
+/**
+ * Executor with an execution count limit.
+ *
+ * Guarantees the number of executions (irrelevant of the Runnable), as
+ * one may wish for this executor to execute at most n times.
+ */
+class ExecuteN
+/** @param limit maximum number of times this executors will [execute] */
+(private val max: Long) : Executor {
+
+    private var executed: Long = 0
+
+    override fun execute(r: Runnable) {
+        if (executed < max)
+            r.run()
+        executed++
```

Comment:
Note: This solution would not occur to me. Of course it works this way too. However it is generally best to avoid returning from method prematurely. It works for preconditions (arg checks), but other than that it gets confusing fast.